extras/package/ubuntu/qmmp/debian-wily/qmmp.1
.\"                                      Hey, EMACS: -*- nroff -*-
.\" First parameter, NAME, should be all caps
.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
.\" other parameters are allowed: see man(7), man(1)
.TH QMMP 1 "July 23, 2008"
.\" Please adjust this date whenever revising the manpage.
.\"
.\" Some roff macros, for reference:
.\" .nh        disable hyphenation
.\" .hy        enable hyphenation
.\" .ad l      left justify
.\" .ad b      justify to both left and right margins
.\" .nf        disable filling
.\" .fi        enable filling
.\" .br        insert line break
.\" .sp <n>    insert n+1 empty lines
.\" for manpage-specific macros, see man(7)
.SH NAME
qmmp \- program to play audio files
.SH SYNOPSIS
.B qmmp
.SH DESCRIPTION
This manual page documents briefly the
.B qmmp
command.
.PP
.\" TeX users may be more comfortable with the \fB<whatever>\fP and
.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
.\" respectively.
\fBqmmp\fP is a program that plays audio files.
.SH AUTHOR
qmmp was written mainly by Ilya Kotov <forkotov02@hotmail.ru>.
.PP
This manual page was written by Eugene V. Lyubimkin <jackyf.devel@gmail.com>,
for the Debian project (but may be used by others).


ChangeLog
Changelog of Qt-based Multimedia Player

--------------
Version 0.0.1
* first test release

Version 0.0.2
* fixed bug in FLAC plugin
* fixed fft.c: changed g_malloc() to malloc()
* fixed UTF bug in FLAC plugin
* fixed UTF bug in Ogg Vorbis plugin

Version 0.0.3
* added tray support (based on Pavel's Kirpichev's patch)
* added menu and hot keys support
* added directory $HOME/.qmmp/skins in skin search paths
* fixed UTF bug in FLAC plugin (previous correction doesn't fix bug)
* fixed UTF bug in Ogg Vorbis plugin (previous correction doesn't fix bug)
* fixed bug in playlist which can cause program crash
* add drag&drop support (thanks to Vladimir Kuznetsov)

Version 0.0.4
* added shift insertion support (Vladimir Kuznetsov)
* added inserted songs moving (Vladimir Kuznetsov)
* added queue support (Vladimir Kuznetsov)
* added sort menu (Vladimir Kuznetsov)
* added shuffle and repeat functions (Vladimir Kuznetsov)
* improved playlist slider
* added volume and balance control
* fixed some bugs in mad plugin
* various improves in plugin system

Version 0.0.5
* added playlist indicators (Vladimir Kuznetsov)
* added playlist control buttons (Vladimir Kuznetsov)
* improved shuffle and repeat functions (Vladimir Kuznetsov)
* added support for m3u, pls, xspf playlists (Vladimir Kuznetsov)
* added region.txt file support
* added scroll control for equalizer bars
* added autosave of equalizer settings
* added Jack support plugin (Yuriy Zhuravlev)
* added Musepack support plugin
* fixed program crush with some skins
* fixed bug in balance control
* fixed bug in alsa plugin which cause crush in some soundcards (thanks to Vadim Kalinnikov)
* fixed bug in processing of double click on playlist

Version 0.0.6
* improved insertion (Vladimir Kuznetsov)
* added command line support (Vladimir Kuznetsov)
* added playlist loading by thread (Vladimir Kuznetsov)
* added equalizer preset support
* added ffmpeg support plugin for playing WMA files
* fixed bug in mad plugin, which cause program crash with invalid file paths
* fixed bug in Jack plugin, which cause program crash when Jack server is
  not running (Yuriy Zhuravlev)
* added pkgconfig support for building

Version 0.1
* added jump to file dialog (Vladimir Kuznetsov)
* added about dialog (Vladimir Kuznetsov)
* added Russian translation
* added installation support
* added cmake support(experimental)
* added default skin
* improved settings dialog
* fixed "assertion failed" bug
* fixed hotkeys in playlist menu
* added English readme (Vladimir Kuznetsov)
* fixed showing bitrate over 999 kbps
* added Winamp EQF import
* added rest time show (Vladimir Kuznetsov)
* fixed time indicator blinking (Vladimir Kuznetsov)
* fixed "segmentation failed" with some buggy skins (Vladimir Kuznetsov)
* fixed parsing some skins

Version 0.1.1
* fixed ffmpeg plugin build
* fixed playlist file types (Vladimir Kuznetsov)
* fixed zero duration bug in mad plugin
* fixed codepage in about dialog
* fixed bug in cmake build scripts, which cause conflict in translation resources

Version 0.1.2
* fixed "segmentation failed" in remaining time mode (Vladimir Kuznetsov)
* fixed "segmentation failed" when resume playback with empty playlist
* fixed settings saving in tray mode
* added hotkeys in playlist (Up, Down, Alt- and Shift- modifiers) (Vladimir Kuznetsov)
* added broken files skipping (Vladimir Kuznetsov)
* added Turkish translation (Mustafa GUNAY)
* added custom close action (Vladimir Kuznetsov)
* fixed and improved cmake scripts
* fixed "Delete" hotkey in Russian translation

Version 0.1.3
* added Chinese Simplified translation (lon)
* added Ukrainian translation (Gennadi Motsyo)
* added codings fix for command line (Vladimir Kuznetsov)
* fixed next song selection after deletion selected (Vladimir Kuznetsov)
* fixed zero duration bug in mpc, flac, vorbis plugins
* fixed some non-critical bugs in Ogg Vorbis plugin
* added Shift+N hotkey for all windows
* set using UTF-8 by default in mad plugin
* fixed Russian translation in the flac plugin
* added current song position navigation with Left/Right cursor buttons (Vladimir Kuznetsov)
* added main menu popup on main display (Vladimir Kuznetsov)
* added menu for left title button (Vladimir Kuznetsov)
* added custom library directory (for qmake only)

Version 0.1.3.1
* fixed library install path when using qmake

Version 0.1.4
* redesigned command line interface (Vladimir Kuznetsov)
* added packed skin support
* xdg menu support (thanks to Gennadi Motsyo and Karel Volný)
* added custom library path
* added current song saving
* removed wrapper from installation
* added Czech translation (Karel Volný)
* added Chinese Traditional translation (lon)
* updated Chinese Simplified translation (lon)
* updated Turkish translation (Mustafa GUNAY)
* fixed Russian Translation
* updated icons
* fixed cmake scripts (Yuriy Zhuravlev)
* reduced wakeups number
* fixed interface freezing when sniping broken entries
* fixed skin parsing
* fixed segmentation fault when trying to show details dialog with invalid file path
* fixed codepage saving in the mpeg plugin
* fixed some bugs in the ffmpeg plugin
* added qt4.3-specific fixes
* fixed metadata disabling
* fixed freezing when seeking

Version 0.1.5
* updated Czech translation (Karel Volný)
* added Brazilian Portuguese translation (Klaos Lacerda and Bruno Gonçalves)
* fixed memory leak
* fixed build with Qt-4.2
* fixed file permissions
* faster seek speed while using alsa plugin as output
* changed playlist background color
* fixed freezing

Version 0.1.6
* fixed desktop file
* fixed Russian translation
* added "$PREFIX/share/qmmp/skins" to skin search paths
* fixed skin parsing
* fixed possible GPL violation
* tar.bz2 skins support
* using ISO-8859-1 encoding for id3v1 tags by default
* fixed load playlists with double extensions
* added German translation (Stefan Koelling)
* improved settings dialog
* added localization for standard Qt dialogs
* show more debug information

Version 0.2.0

src/qmmp/qmmp.cpp
/***************************************************************************
 *   Copyright (C) 2008-2013 by Ilya Kotov                                 *
 *   forkotov02@hotmail.ru                                                 *
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This program is distributed in the hope that it will be useful,       *
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
 *   GNU General Public License for more details.                          *
 *                                                                         *
 *   You should have received a copy of the GNU General Public License     *
 *   along with this program; if not, write to the                         *
 *   Free Software Foundation, Inc.,                                       *
 *   51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.         *
 ***************************************************************************/

#include <QDir>
#include <QApplication>
#include <QSettings>
#include <QLocale>
#include <QByteArray>

#ifndef LIB_DIR
#define LIB_DIR "/lib"
#endif

#define DEV_SUFFIX "dev"

#include "qmmp.h"

QString Qmmp::m_configDir;
QString Qmmp::m_langID;

const QString Qmmp::configFile()
{
    return m_configDir + "qmmprc";
}

const QString Qmmp::configDir()
{
    return m_configDir.isEmpty() ? QDir::homePath() +"/.qmmp/" : m_configDir;
}

void Qmmp::setConfigDir(const QString &path)
{
    m_configDir = path;
    if(!m_configDir.endsWith('/'))
        m_configDir.append('/');
}

const QString Qmmp::strVersion()
{
    QString ver = QString("%1.%2.%3")
            .arg(QMMP_VERSION_MAJOR)
            .arg(QMMP_VERSION_MINOR)
            .arg(QMMP_VERSION_PATCH);
#if !QMMP_VERSION_STABLE
#ifdef SVN_REVISION
    ver += "-svn-"SVN_REVISION;
#else
    ver += "-"DEV_SUFFIX;
#endif
#endif
    return ver;
}

const QString Qmmp::pluginsPath()
{
    QByteArray path = qgetenv("QMMP_PLUGINS");
    if (!path.isEmpty())
       return path;
#ifdef QMMP_INSTALL_PREFIX
    QDir dir(QMMP_INSTALL_PREFIX"/"LIB_DIR"/qmmp");
    //qDebug(QMMP_INSTALL_PREFIX"/"LIB_DIR"/qmmp");
#else
#if defined(Q_OS_WIN) && !defined(Q_OS_CYGWIN)
    QDir dir(qApp->applicationDirPath() + "/plugins");
#else
    QDir dir(qApp->applicationDirPath() + "/../"LIB_DIR"/qmmp");
#endif
#endif
    return dir.canonicalPath();
}

QString Qmmp::systemLanguageID()
{
    if(m_langID.isEmpty())
    {
        m_langID = uiLanguageID();
        //qDebug("Qmmp: setting ui language to '%s'", qPrintable(m_langID));
    }

    if(m_langID != "auto")
        return m_langID;

#ifdef Q_OS_UNIX
    QByteArray v = qgetenv ("LC_ALL");
    if (v.isEmpty())
        v = qgetenv ("LC_MESSAGES");
    if (v.isEmpty())
        v = qgetenv ("LANG");
    if (!v.isEmpty())
        return QLocale (v).name();
#endif
    return  QLocale::system().name();
}

QString Qmmp::uiLanguageID()
{
    QSettings settings(Qmmp::configFile(), QSettings::IniFormat);
    QString langID = settings.value("General/locale", "auto").toString();
    langID = langID.isEmpty() ? "auto" : langID;
    return langID;
}

void Qmmp::setUiLanguageID(const QString &code)
{
    QSettings settings(Qmmp::configFile(), QSettings::IniFormat);
    settings.setValue("General/locale", code);
    m_langID.clear();
}
